external/CMakeLists.txt

```diff
@@ -54,6 +54,16 @@ ExternalProject_Add(noise-c
 add_library(noiseprotocol STATIC IMPORTED GLOBAL)
 set_property(TARGET noiseprotocol
   PROPERTY IMPORTED_LOCATION ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libnoiseprotocol.a)
+set_target_properties(noiseprotocol
```

Comment:
isn't this the same as target_include_direcetories with the INTERFACE keyword?

https://stackoverflow.com/questions/26243169/cmake-target-include-directories-meaning-of-scope

Reply:
yeah but that isn't allowed for imported libraries. https://cmake.org/cmake/help/v3.10/command/target_include_directories.html

> The named <target> must have been created by a command such as add_executable() or add_library() and must not be an IMPORTED target.